include/oneapi/tbb/detail/_flow_graph_impl.h

```diff
@@ -359,24 +360,21 @@ class graph : no_copy, public graph_proxy {
 
 };  // class graph
 
+template<typename DerivedType>
 inline void graph_task::destruct_and_deallocate(const execution_data& ed) {
     auto allocator = my_allocator;
     // TODO: investigate if direct call of derived destructor gives any benefits.
     this->~graph_task();
```

Comment:
Maybe we will resolve the TODO? Check that all derived are final (does it make sense to mark them as `final`)? Or you can address it in a separate PR.
```suggestion
    this->DerivedType::~graph_task();
```